doc/services/pm/evtc.rst

```diff
@@ -0,0 +1,189 @@
+   * The maximum acceptable latency of the EVT.
+3. PM EVTC minimizes EVT latency just in time for it to be in
+   effect before the event is predicted to originate.
+4. EVT originates
+5. EVT is handled by the user
+6. User releases scheduled PM EVT using
+   :c:func:`pm_evtc_release_evt()`
+7. PM EVTC maximizes EVT latency.
+
+The time spent with minimized latency, and thus higher energy
+consumption, is minimized to just before the event to just after
+the event has been handled, as notified by the user.
+
+A PM EVT can be additionally be rescheduled using
+:c:func:`pm_evtc_reschedule_evt()` which implicitly and efficiently
+releases the PM EVT followed by scheduling it.
+
+Unpredictable EVT latency management
+************************************
+
+Purpose
+-------
+
+To ensure a maximum EVT latency while maximizing EVT latency.
+
+Usage
+-----
+
+1. User configures EVT originator to originate EVTs
+2. User requests maximum EVT latency with PM EVTC using
+   :c:func:`pm_evtc_request_evt()`.
+3. PM EVTC minimizes EVT latency immediately to be in effect as "soon
+   as possible".
+
+While the EVT latency is in effect, it is not neccesarily constant,
+but will not exceed the requested maximum.
+
+4. EVT originates.
+5. EVT is handled by the user.
+
+The above steps are repeated until the user no longer requires a
+maximum EVT latency.
+
+6. User releases EVT using :c:func:`pm_evtc_release_evt()`.
+7. PM EVTC maximizes EVT latency.
+
+PM EVTC implementation
+**********************
+
+Devicetree
+----------
+
+Event controllers which support power management must include the base
+binding ``evtc.yaml``:
+
+.. code-block:: yaml
+
+   include:
+     - "evtc.yaml"
+
+which includes the required properties ``evtc-request-latency-ns`` and
+``evtc-latencies-ns`` which must be defined in the devicetree:
+
+.. code-block:: devicetree
+
+   evtc0: evtc0 {
+           /* Latency for request to take effect in nanoseconds */
+           evtc-request-latency-ns = <3500>;
+           /* Supported event latencies in nanoseconds */
+           evtc-latencies-ns = <200 150 50 20>;
+   };
+
+Device driver
+-------------
+
+PM EVTC device drivers must implement the API
+:c:func:`pm_evtc_request()` to which the following contract applies:
+
+* If :kconfig:option:`CONFIG_PM_EVTC` ``=y`` the highest supported
+  event latency will be requested on init.
+* If :kconfig:option:`CONFIG_PM_EVTC` ``=n`` the lowest supported
+  event latency will be requested on init.
+* Only supported latencies will be requested as defined in the
+  devicetree property ``evtc-latencies-ns``
+* Only latencies different from the current latency will be requested.
+* Request intervals are limited by the devicetree property
+  ``evtc-request-latency-ns``.
+* The request will be performed from ISR context, and is not reentrant.
+
+The PM EVTC instance which implements the :c:func:`pm_evtc_request()`
+API is defined using the :c:macro:`PM_EVTC_DT_DEFINE()`:
```

Comment:
Is it generic / supposed to be implemented by any device driver ?  Or only "event controller " devices ? If that is not generic for any type of device why don't have it implemented as new type of driver instead of part of pm subsystem ?

Reply:
Any device which deals with events, comparators, gpio egde detectors, CPU irq handling latencies, etc.